benchmarks/perception/a3_stereo_image_proc/benchmark.yaml

```diff
@@ -0,0 +1,85 @@
+  A simple perception computational graph composed by 2 Components, `rectify` and `resize` operations. Used to demonstrate a simple perception pipeline using the  package.
+short: Perception computational graph to compute a disparity map for stereo images.
+graph: /imgs/a3_stereo_image_proc_graph.png
+metric:
+  metric: "latency"
+  unit: "ms"
+reproduction: |
+  ## ------- Installation -------
+  # Create a ROS 2 overlay workspace
+  mkdir -p /tmp/benchmark_ws/src
+
+  # Clone the benchmark repository
+  cd /tmp/benchmark_ws/src && git clone https://github.com/robotperf/benchmarks
+
+  # Fetch dependencies
+  source /opt/ros/humble/setup.bash
+  cd /tmp/benchmark_ws && sudo rosdep update || true && sudo apt-get update &&
+    sudo rosdep install --from-paths src --ignore-src --rosdistro humble -y
+
+  # Install the stereo_image_proc package
+  sudo apt-get install ros-humble-stereo-image-proc
+
+  # Build the benchmark
+  colcon build --merge-install --packages-up-to a3_stereo_image_proc
+
+  # Terminal 1: Replay rosbag in a loop (within rosbags repo)
+  cd /path/to/rosbags/perception
+  ros2 bag play --loop image3
+
+  # Verify you are working in the workspace
+  cd /tmp/benchmark_ws
+
+  # Source the workspace as an overlay
+  source install/setup.bash
+
+  # Terminal 2: View Graph
+  rqt_graph
+
+  # Terminal 3: Visualize Image Feed
+  rviz2 -d src/benchmarks/benchmarks/perception/a3_stereo_image_proc/config/camera_bot.rviz
+
+  # Terminal 4: Test Stereo Image
+  ros2 launch a3_stereo_image_proc trace_a3_stereo_image_proc.launch.py
+
+  # Terminal 5: View Stereo Image Disparty Map
+  ros2 run image_view disparity_view image:=/benchmark/disparity
+
+  # Verify data was collected
+  cd ~/.ros/tracing
+  babeltrace a3_stereo_image_proc | less
+
+  ## ------- Reproduce Simulation -------
+  cd /tmp/benchmark_ws
+
+  # Source
+  source install/setup.bash
+
+  # Terminal 1: Launch Simulation
+  cd /tmp/benchmark_ws 
+  ros2 launch a3_stereo_image_proc launch_sim.launch.py world:=src/benchmarks/benchmarks/perception/a3_stereo_image_proc/worlds/neighborhood.world
+
+  # Terminal 2: Visualize perception from robot's perspective using rviz
+  rviz2 -d benchmarks/perception/a3_stereo_image_proc/config/camera_bot.rviz
+
+  # Terminal 3: Record a rosbag (ctrl+c to end recording)
+  ros2 bag record -a
+
+  # Terminal 4: Use keyboard to drive robot around world
+  ros2 run teleop_twist_keyboard teleop_twist_keyboard
+
+  # Verify: Play back rosbag after recording it in a loop
+  ros2 bag play --loop /path/to/bag/directory/
```

Comment:
Are there guidelines around how complex the launch process should be for benchmarks?  Having to launch `N` (=4 or 5, in this case) is labor-intensive.  Couldn't a convenience script be provided for the end-user to spawn all processes at once?

Reply:
Yeah, that makes sense! I just added a script that will spawn all the optional processes at once!